src/backend/optimizer/plan/plangroupext.c

```diff
@@ -1858,13 +1911,34 @@ generate_list_subplans(PlannerInfo *root, int num_subplans,
 	return subplans;
 }
 
-static void
-destroyGroupExtContext(GroupExtContext *context)
+/*
+ * Generate a list of simple_rel_array 
+ */
+static List *
+generate_list_simple_rel_array(PlannerInfo *root, int num_subplans)
 {
-	if (context->sub_tlist)
-		list_free(context->sub_tlist);
-	if (context->canonical_rollups)
-		list_free_deep(context->canonical_rollups);
+	int i;
+	int j;
+	List *relArrayList = NULL;
+
+	for (i = 0; i < num_subplans; i++)
+	{
+		int arraySize = root->simple_rel_array_size;
+		RelOptInfo **relArray = (RelOptInfo **) palloc0(sizeof(RelOptInfo *) * arraySize);
+		for (j = 0; j < arraySize; j++)
+		{
+			RelOptInfo *rel = NULL;
+			if (root->simple_rel_array[j] != NULL)
+			{
+				rel = makeNode(RelOptInfo);
+				memcpy(rel, root->simple_rel_array[j], sizeof(RelOptInfo));
```

Comment:
Is `memcpy` enough here?  Should we consider to use `copyObject` instead?

Reply:
we can't do copyObject on RelOptInfo. I think memcpy should be fine here.